### PR TITLE
Support multiselect in ObjectSearch

### PR DESCRIPTION
**Pull request by mumumumu**
- adds support for multiselect in ObjectSearch.
- also adds `onChange` to `IObjectSearchProps` to fix https://github.com/mighty-justice/fields-ant/issues/240

object-search-multiselect

### CODE REVIEW

File: src/inputs/ObjectSearch.tsx [hunk 1]
```diff
@@ -221,8 +228,15 @@ class ObjectSearch extends Component<IObjectSearchProps> {
     }
 
     // Select from search
-    const foundOption = this.options.find(option => option.id === selectedOption.key);
-    onChange(toJS(foundOption));
+    if (this.isMultiSelect) {
+      const foundOptions = this.options.filter(option => (
+        selectedOption.map((_selectedOption: any) => _selectedOption.key).includes(option.id)
```
Comment (SHA-4) [inline src/inputs/ObjectSearch.tsx]: I'd split this map into a constant for clarity

Comment (rkuykendall) [inline src/inputs/ObjectSearch.tsx]: Might be nice to move this map outside of the filter loop, but this function isn't called on render so it's not a huge deal.

File: test/types/objectSearchCreate.test.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ function getDefaults (overrides?: any) {
   return {
     expectedLabel: 'Law Firm',
     fakeOwed: faker.finance.amount(),
-    result: fakeLawFirm(),
+    results: Array(3).fill(null).map(() => fakeLawFirm()),
```
[the remark this answers is not in the document]
Comment (mumumumu) [reply test/types/objectSearchCreate.test.ts]: No, but I can make factory for fake organization results.

File: src/inputs/ObjectSearch.tsx [hunk 3]
```diff
@@ -97,6 +99,11 @@ class ObjectSearch extends Component<IObjectSearchProps> {
     return !this.hasOptions && !this.hasSearch;
   }
 
+  private get isMultiSelect () {
+    const { mode } = this.selectProps;
+    return mode && ['multiple', 'tags'].includes(mode);
```
[the remark this answers is not in the document]
Comment (mumumumu) [reply src/inputs/ObjectSearch.tsx]: I get this error message with the `mode &&`:
```
ERROR in [at-loader] ./src/inputs/ObjectSearch.tsx:104:42
    TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.
```

**Pull request merged**